### ISSUE TITLE
[Suggestion] Walk/running speed lock

### ISSUE THREAD

**Issue by ZachZent** [tsunamayo/Skywanderers#1784]
![](https://cdn.discordapp.com/attachments/271723356795961345/702327373407977542/Sass.gif)
In game when in "None" locomotion, you can still change the speed with the brackets. That creates situations like the one in the trailer with the sassy cat strutting on the bridge. 
![](https://cdn.discordapp.com/attachments/406664114249793536/702335564803866634/unknown.png)
For "None" and Jetpack, I suggest locking in a single walk and run speed while keeping the option to go slower or faster for Anti-gravitation. Full 6 dof for now should stay as is.

Just as a reference, Minecraft's walking speed is 4.3 m/s and the running speed is 5.6 m/s.

**Comment by tsunamayo** [comment]
Yes sassy cat got some attention - I will have to address this ;). Also all my animation are from a female character, I need to change that...
Thanks for the minecraft speed!
So I wanted to get rid of the [ ] for normal game.
I will keep them as a debug option only.
But I wanted to have press SHIFT to boost in anti grav and 6DOF, possible multiple time for really significant boost. 
I will experiment a bit with that.

**Comment by ZachZent** [comment]
I'd say that anti-grav should only be for creative and shipyards since it really does make all other forms of movement, especially 6 dof, obsolete.

As for 6 dof I like the idea of pressing shift to speed up, though this should be at a cost of fuel similar to SE.

**Comment by tsunamayo** [comment]
I like the idea of making 6dof obsolete, honestly it is there for some very specific case and because SE players where crying about this...
It really kills me every time I try to watch a SE vid and I see player mis-aligning themselves constantly when in a ship or on the ground...
By the way 6dof is also anti-grav.
Yes fuel could be thing once I get to add more survival elements... Also I would need a proper model for the jetpack and the helmet...
Cheers

**Comment by ZachZent** [comment]
In that case I'd say improving 6 dof to be more accommodating instead of two separate ones. 
No matter what you do anti-grav alone will be too powerful or too under powered.
* OP
Creative mode in Minecraft at all times where you can fly over walls except in SEVO where you can just take off to the enemy ship dodging all the way
* UP
Adding a limiter like fuel to anti-grav means that building will be super hard as you need to constantly refuel. And if you don't as the fuel capacity is large then you get the OP problem again

Make maneuvering tighter than SE, magnet shoes for latching onto ships, and speed control will be the best of both worlds

**Comment by tsunamayo** [comment]
Hum I see. I dont want to get rid of ag, 6dof will always have the mis-alignment issue.
I feel that when you just want to build this should not be a pain to move around your ship, even in survival.
I think we need more brainstorming! Not really urgent but no arm in starting now.
What about a ag jetpack that feeds itself on the ship main generator?

**Comment by ZachZent** [comment]
That does get rid of the problem of flying off anywhere you go, but it is still a problem with players just flying around in virtually creative mode. Imagine getting boarded and someone is just floating 6 feet off the ground dodging every which way. Or even worse both players doing such a thing as a ship is still a ship regardless of the owner.
![](https://i.imgflip.com/3xmozc.jpg)

You are right that the 6 dof causes misalignment, but I feel combining the two into one that doesn't have the godlike ability of anti-grav/creative mode and isn't awkward like SE 6dof is the best option. Especially since you said "I like the idea of making 6dof obsolete." Why have something obsolete in game when you can improve it. Besides making the turning tighter/less inertia, fuel, etc, one way to fix it is adding back the alignment option like we used to have. Middle click -> align. That way players can flip any which way, which is a good thing, but can always come back to the orientation of the ship.

But since you probably don't want to add the middle click option back, why not integrate it into the space suit? You need a space suit to [6dof/anti-grav] but some integrated option or hotkey that allows you to realign to an entity without needing to spin every which way. "Magnet shoes" or something.

**Comment by ZachZent** [comment]
There is also the option of adding some sort of building mode similar to Starmade, one of the few good ideas they had. In this build mode you could anti-grav while being out you can only 6dof, walk, jetpack. Maybe to incorporate the drones, your body gets "left behind" while a drone flies around and build. Your body can still be killed preventing a player from going into build mode to escape death.

A bunch of drones players made awhile back
![](https://cdn.discordapp.com/attachments/406664114249793536/702428608479625226/unknown.png)

**Comment by FelixGold** [comment]
Empyrion has a mode where you deploy a drone and leave your body standing there, its a really useful and feature like Zach mentioned, and further it would be AWESOME if we could choose to use a default drone, our lil yellow buddy, or personalize it and build our own. so each person has a choice of their buddy drone looking unique.

**Comment by tsunamayo** [comment]
@ZachZent Funny picture!

First I cant fix 6dof. As soon as you look upward / downward your body tilt and mis-align. Look up, then left, then down to the horizon, and you will be misaligned.
What you are describing is exactly like the ag mode. Instead of middle click or pressing a button, you just switch your mode with the menu. Thats really the same thing, you just activate it in a different manner.
Then I agree 100% for the combat, but that the same things with the 6dof. So I dont really get the whole ag vs 6dof things. They both have the save issue for survival.
Drone idea is sexy, but then what about the multi-tool? There is so much implication I dont really want to go there!

**Comment by FelixGold** [comment]
The drone could also have a multitool, and dpending on how much survival elements you want, like atmosphere, radiation, oxygen and the like it could go into hostile areas, it is also considerably smaller than the player avatar, so it can fit through one block spaces.

**Comment by ZachZent** [comment]
Ok I see. I was more thinking along the lines of inertia rather than just moving. I'm not one of the SE refugees so I'm not exactly used to 6dof.

For combat there is a difference. Anti-grav is sharp turns and instant changes in direction. 6dof would be actually maneuvering. Needing to slow and stop to change directions. It is much harder to shoot someone who can turn on a dime than someone who has consider physics.

You can still have the multi-took for building stuff. Player will be building and fixing all along, but will have to use 6dof. To be able to fly with anti-grav you use a the drone which will be more for the bulk of construction. When survival is in and players spend more of their time flying and exploring they will be using the multi-tool quite a lot. Especially if you go the NMS route for multitools which is one part of that game I like. This removes the sharp turns and instant changes in speed in gameplay, but still allowing players to build comfortably without consequences.

**Comment by tsunamayo** [comment]
Yes the drone idea is really sexy, actually I was toying with that idea recently but for repair drone.
At this stage I think I need repair drone for combat, to fix hull and turret. 
And of course people wants to build their cute little drone! Only problem was the size, I would have to fit in a 1mx1m cube, and I dont like the idea of scaling down brick size...

**Comment by FelixGold** [comment]
Any way it can have a core block, and just not let the build environment for the drone expand beyond the 1mx1m outer limit?

**Comment by ZachZent** [comment]
Of the drones I posted, most are smaller than 1x1x1. Minus the antennas of course. The brick size doesn't need to be scaled down. To quote a comment on a very old post of mine talking about a smaller brick grid for weapon crafting

> I wont do that but yes one of the reason I didnt release the weapon craft is because I need more interesting looking bricks.

At worst the drone can be a prefab, but the player controlled drone builder idea would solve the overall problem to allow anti-grav for comfortable building but not for general gameplay. 6dof would still let you fly, but with limits such as fuel and physics constraints

**Comment by Kaiser-Indrasil** [comment]
On the matter of 6dof vs. ag
In immersive games like Star Citizen mis-alignment in 6dof is a powerful tool to convey the confusing nature of 3-dimensional space. It's made for immersion. To make you feel lost and small.
It also can be used to control the camera in a more unencumbered way. This could prove useful for machinima makers.

My proposed solution, is to merge the gravity align and 6dof modes, but also, to let players switch between them with a simple keybind. There surely is a key not currently being used in game that we could use for that.

As a salty Space Engineers refugee, I actually do like the jetpack in that game. Well, somewhat. While I like the idea of inertia conservation, I don't share the same enthusiasm for the speed at which SE lets you fly using jetpack. 110m/s of maximum speed with over 15m/s^2 of acceleration is broken and feels highly unrealistic. On the other hand, the flightsuit thrusters in SC let you fly at a mind-blowing speed of... 3m/s. Maybe not exactly, but it feels *really* slow. The jetpack in Kerbal Space Program lets you accelerate at a pace of roughly 5m/s^2 but with an uncapped top speed... at least until your fuel goes empty or you die in an atmospheric re-entry.

While in KSP there is no concept of inertia dampeners (auto-breaking when no player movement input is detected), there is a toggle for it in SE, and in SC they are always on. I think SE wins that one.

Since we have to consider how things would look like in the survival mode of Starship EVO, we have to come up with 2 ways of handling building. My idea is to use both the multi-tool, and a player-made drone. Here's how:

> Limit the speed of player's jetpack* to 10 meters per second. That would be a uniform value both in survival and creative modes. Make the acceleration 5m/s^2. How would the player move quickly and efficiently in the creative mode? That's where the drone comes into play.
The drone would be powered by the proximity to the parent grid. There would be one building/destroying drone per grid. It could even have its little hangar. <3 Much like the game calculates the shape of the shield, it could calculate the drone's movement sphere, which would obviously need to be bigger. The drone would have its max speed and acceleration 10 times bigger than the player's. Its effective speed and acceleration caps, however, could be regulated by keybinds (modifier like Shift / Ctrl / Alt + scroll wheel) to let the player adjust the drone's quickness on the fly.

The idea of the drone further aids with the creation of machinimas since it can record the player themselves.

*Since we're EVOlution of other space building games, how about something more futuristic than a jetpack? Maybe some sexy small anti-grav propellers here and there - think Genji's green circle things on his armour. Maybe even recolorable? Is that too much to ask? ;P

![image](https://user-images.githubusercontent.com/63756057/79965073-7c58d480-8483-11ea-83fb-50f1cd65c159.png)

**Comment by tsunamayo** [comment]
Thanks for the detailed feedback! I like the armor very much, indeed that could be something more original than a jet pack, I had in mind an iron-man suit also.
As you are a SE refugee ;), just be aware that I do not wish to go in the lets use every key of the keyboard route. I only resort to key when it makes sense, I try to minimize the learning curve.
This is why I put different mode - I dont think that changing locomotion mode is a high frequency action, hence it can be move to a menu.
I still think that the easiest solution will be to give AG in creative or in shipyard for survival.
Or you could get a block that make a big AG bubble around your ship and then use that mode as you wish. So no AG for exploring or boarding other ships.

**Comment by ZachZent** [comment]
>  I still think that the easiest solution will be to give AG in creative or in shipyard for survival.
That I think is fair at least at the start. In creative mode it doesn't matter and shipyards are shipyards which is fine. However if you can clarify something. How do you want us to build ships? Not just place an extra block or two and repair, but truly construct them? Will we have to go to a shipyard to build ships or can we start and build a ship anywhere? Or do you have to go to a shipyard to start a ship, but can build anywhere?

If you can only start a ship and build in a shipyard then problem solved. Keep anti-grav only around a shipyard.
If you can build both in a shipyard or start and build a ship in the middle of nowhere equally then keeping anti-grav in a shipyard only while 6 dof can be used everywhere else for building will also solve it. 
Keeping the quick, turn on a dime creative mode flight as an option even limited to your ship still has problems in my opinion and should be avoided.

If you can start and build a ship anywhere and shipyards are used for just spawning blueprints and repairing ships then I would still advocate for the drone idea. Long term of course, totally doesn't need to be changed before EA or shortly after. Combines the accessibility of building without giving any player any sort of advantage even on their own ship. Also I'm imagining everyone drone is in their suit and would launch at a press of a button. It could have multiple uses like scouting.

And yes, Genji space suit armor please and thank you

**Comment by tsunamayo** [comment]
So for survival I wanted it on shipyard. It might be near the shipyard, like around the station.
The reason is that you will pay some cash to buy the block, so it is more about getting the blocks.
Then not sure for long term, as I want to move toward full mining / crafting. I still think paying cash will be needed, some sort of insurance cost as your ship get free repair and free respawn, but it will be lower to compensate.
What do you think I should do? Ultimately I dont see why I should restrict player around shipyard once I get the full loop.

**Comment by FelixGold** [comment]
No reason you couldn't have both? If the players want to build with cash they have to do it around or on a shipyard as you are building from their inventory, meanwhile if a player wants to mine and produce their own blocks they can do it on the fly.

That way players can do what they like, and it produces player driven emergent gameplay, where players that mine and build and sell can then use the cash they get from that to hire players who don't wanna do the mining and stuff themselves?

**Comment by tsunamayo** [comment]
yeah thats what I meant, I just said the very first loop wont have  mining, just cash. I have to take it a step at a time

**Comment by ZachZent** [comment]
Based on what you said, I believe the best option in the distant future is that you can start a ship anywhere and build anywhere with a starter block, however you will only have the blocks you have on hand. I don't believe it will be much of a problem for the most part. Most of the "building" in the null when survival elements are fully added will be smaller modifications and patchwork. Perhaps adding a new turret or part of the ship to accommodate something new. And for interior work you will have the gravity block to keep you upright inside the ship. 6 dof wouldn't be too much of a hassle for this. 

At shipyards and the areas around it is where you will be able to use the anti-grav to build. This includes ground yards as well as space yards. This can be the player using anti-grav, but I would still suggest a player controlled drone while the player's body stays where it is. It makes more sense that a drone would be able to fly every which way in space and on planets compared to Mr. player. Plus the player wouldn't need to worry about say oxygen to build when in space yard. The shipyard would also give you access to their block inventory to supplement the blocks you will have on hand. This of course would be later in survival development.

As for insurance this should totally be a thing. One of the corporate moguls, I think Xerma, was going to make their corporation a bank. An ATM could be at each station where players can deposit and withdraw money (clips.) This could also double as a way to buy insurance for your ship where you buy insurance from the ATM.

**Comment by tsunamayo** [comment]
haha a bank yes! I am totally for all sort of greedy corps. I have been thinking about high values goods transportation lately, using dwarf-lord system #1292.

**Comment by ZachZent** [comment]
Well I still plan to do ARX, the prison corp. I'd suggest making a list of all the industries you want in game that can be made by the Corporate Moguls and above. Like bank, mail service, etc. Give some of the CMs ideas of what you want so we can build for the in game systems.

And of course quests for said corps. Raiding a banking fleet transporting lots of money. Raid the prison ship to get more crew or rescue someone.
